prost-build/src/code_generator.rs

```diff
@@ -44,6 +44,45 @@ fn push_indent(buf: &mut String, depth: u8) {
         buf.push_str("    ");
     }
 }
+
+struct Field {
+    rust_name: String,
+    descriptor: FieldDescriptorProto,
+    path_index: i32,
+}
+
+impl Field {
+    fn new(descriptor: FieldDescriptorProto, path_index: i32) -> Self {
+        Self {
+            rust_name: to_snake(descriptor.name()),
+            descriptor,
+            path_index,
+        }
+    }
+}
+
+struct OneofField {
+    rust_name: String,
+    descriptor: OneofDescriptorProto,
+    fields: Vec<(FieldDescriptorProto, i32)>,
```

Comment:
Should this be typed as `Vec<Field>`?

Reply:
Good idea, since `Field` will be just these data.